lib/cocoapods/generate/installer.rb

```diff
@@ -114,7 +114,7 @@ def create_app_project
 
         spec_platforms
           .map do |platform|
-            consumer = spec.consumer(platform)
+            consumer = specs.first.consumer(platform)
```

Comment:
I think this might be a mistake. We should not assume the first spec has the platform we expect. Its possible you get two specs one for iOS and one for osx for example and I think this will break. Will fix and add a test.

Reply:
ah lol there are no tests for this class...

Reply:
confirmed my intuition that this will crash with a test. Fixing it.

Reply:
added tests and fixed logic!